hack/generator/spec/crossplane.md

```diff
@@ -0,0 +1,71 @@
+# Thoughts on performing provider-azure code generation for Crossplane
+
+The patterns identified below were extrapolated from [the Crossplane VNET spec](https://github.com/crossplane/provider-azure/blob/master/config/crd/network.azure.crossplane.io_virtualnetworks.yaml).
+
+## General notes
+
+### Crossplane has the following static "special" properties
+  1. ClaimRef [Deprecated]
+  2. ClassRef [Deprecated]
+  3. DeletionPolicy
+  4. ProviderConfigRef
+  5. ProviderRef [Deprecated]
+  6. ReclaimPolicy [Deprecated]
+  7. writeConnectionSecretToRef
+
+Since these are statically shaped they should be relatively easy to add with Crossplane specific generator pipeline stages.
+
+### Parent references in Crossplane are really a discriminated union of 3 options:
+1. XName (e.g. ResourceGroupName)
+2. XNameRef (e.g. ResourceGroupNameRef)
+3. XNameSelector (e.g. ResourceGroupNameSelector)
+
+We probably will eventually want to understand how `XName` differs from `XNameRef` (I think I understand the `XNameSelector`). 
+
+Generating these might take a bit of work as we need to derive the name(s) for each of these for each of the parent resources. This should be doable in a pipeline stage by walking up the graph of owners that we already have and getting their names, then generating a set of these for each owner.
+
+### Status 
+Our status's include the "full" shape of the object, whereas Crossplane's seem to be limited (they don't duplicate the spec, or include other server-derived fields). We should ask if this is an intentional design decision or just something they did in the interest of time.
```

Comment:
Per out of band conversation, this decision was intentional though somewhat controversial. :)

There' some context on how we decide what goes into spec vs status in our [API design doc](https://github.com/crossplane/crossplane/blob/a3425d5a/design/one-pager-managed-resource-api-design.md#high-fidelity). In general if an API caller can ever set a field it goes into `spec`, and if they can never set a field it goes into `status`. Spec fields that a caller _could_ set but that are defaulted by Azure when left unset are [late initialized](https://github.com/kubernetes/community/blob/b43d5a2/contributors/devel/sig-architecture/api-conventions.md#late-initialization).

This does mean it's possible that our desired state could be different from actual state and we won't show it, per https://github.com/crossplane/crossplane/issues/925. Including the entire resource in status might help this, but it seems unconventional for Kubernetes resources to do so.

Reply:
Thanks for this clarification. I've updated the document some with this information.